### ISSUE TITLE
[bug] Invalid defines in Bazel build file when the define is a string type

### ISSUE THREAD

**Issue by mohabouje** [conan-io/conan#10563]
Running into this issue on the latest commit in dev branch when trying to use Dear ImGUI:

```
    defines = [
        "IMGUI_USER_CONFIG='imgui_user_config.h'"
    ],
```
That's an invalid definition in Bazel. 

The expected (works fine for me, probably there are other ways to do this) is:
```
    defines = [
        "IMGUI_USER_CONFIG=\\\"imgui_user_config.h\\\""
    ],
```

For now, it's not a big deal as I end up doing:

```
    defines = [
        "IMGUI_USER_CONFIG=\\\"imconfig.h\\\""
    ],
    copts = ["-w"],
```

disabling the GCC warnings.

The fix seems trivial when exporting defines from the Conan recipe, if it's a string, convert it to this format. 

I will prepare a pull request if I have some time this week.

**Comment by memsharded** [comment]
Hi @mohabouje 

Yes, it seems relatively easy. If you can submit that PR, fantastic, if not, just tell, I think we might be able to try it to fix this for 1.46. Thanks for reporting and offering to help.

**Comment by memsharded** [comment]
This has been fixed by https://github.com/conan-io/conan/pull/10812
